engine/access/rpc/backend/backend_transactions_test.go

```diff
@@ -328,3 +333,197 @@ func (suite *Suite) TestGetTransactionResultUnknownFromCache() {
 	})
 }
+
+// TestGetSystemTransaction_HappyPath returns system transaction using GetSystemTransaction call
+func (suite *Suite) TestGetSystemTransaction_HappyPath() {
+	suite.withPreConfiguredState(func(snap protocol.Snapshot) {
+		suite.state.On("Sealed").Return(snap, nil).Maybe()
+
+		params := suite.defaultBackendParams()
+		backend, err := New(params)
+		suite.Require().NoError(err)
+
+		block := unittest.BlockFixture()
+		blockID := block.ID()
+
+		res, err := backend.GetSystemTransaction(context.Background(), blockID)
+		suite.Require().NoError(err)
+
+		systemTx, err := blueprints.SystemChunkTransaction(suite.chainID.Chain())
+		suite.Require().NoError(err)
+
+		suite.Require().Equal(systemTx, res)
+	})
+}
+
+// TestGetSystemTransactionResult_HappyPath returns system transaction result for required block id
+func (suite *Suite) TestGetSystemTransactionResult_HappyPath() {
+	suite.withPreConfiguredState(func(snap protocol.Snapshot) {
+		suite.state.On("Sealed").Return(snap, nil).Maybe()
+		lastBlock, err := snap.Head()
+		suite.Require().NoError(err)
+		identities, err := snap.Identities(filter.Any)
+		suite.Require().NoError(err)
+
+		block := unittest.BlockWithParentFixture(lastBlock)
+		blockID := block.ID()
+		suite.state.On("AtBlockID", blockID).Return(
+			unittest.StateSnapshotForKnownBlock(block.Header, identities.Lookup()), nil).Once()
+
+		suite.blocks.
+			On("ByID", blockID).
+			Return(block, nil).
+			Once()
+
+		receipt1 := unittest.ReceiptForBlockFixture(block)
+		suite.receipts.
+			On("ByBlockID", block.ID()).
+			Return(flow.ExecutionReceiptList{receipt1}, nil)
+
+		params := suite.defaultBackendParams()
+		connFactory := connectionmock.NewConnectionFactory(suite.T())
+		connFactory.On("GetExecutionAPIClient", mock.Anything).Return(suite.execClient, &mockCloser{}, nil)
+		params.ConnFactory = connFactory
+
+		exeEventReq := &execproto.GetTransactionsByBlockIDRequest{
+			BlockId: blockID[:],
+		}
+
+		eventsPerBlock := 10
+		eventMessages := make([]*entities.Event, eventsPerBlock)
+
+		exeEventResp := &execproto.GetTransactionResultsResponse{
+			TransactionResults: []*execproto.GetTransactionResultResponse{{
+				Events:               eventMessages,
+				EventEncodingVersion: entities.EventEncodingVersion_JSON_CDC_V0,
+			}},
+		}
+
+		suite.execClient.
+			On("GetTransactionResultsByBlockID", mock.Anything, exeEventReq).
+			Return(exeEventResp, nil).
+			Once()
+
+		backend, err := New(params)
+		suite.Require().NoError(err)
+
+		res, err := backend.GetSystemTransactionResult(
+			context.Background(),
+			block.ID(),
+			entities.EventEncodingVersion_JSON_CDC_V0,
+		)
+		suite.Require().NoError(err)
+
+		systemTx, err := blueprints.SystemChunkTransaction(suite.chainID.Chain())
+		suite.Require().NoError(err)
+
+		suite.Require().Equal(flow.TransactionStatusExecuted, res.Status)
+		suite.Require().Equal(systemTx.ID(), res.TransactionID)
+		suite.Require().Equal(convert.MessagesToEvents(eventMessages), res.Events)
```

Comment:
can you add a check that decoding works as well?

something like
```golang
_, err = jsoncdc.Decode(nil, res.Events[0].Payload)
suite.Require().NoError(err)
```

you'll need to add at least one real event. you can generate them with the event generator
https://github.com/onflow/flow-go/blob/a368dde941e00dfc7a153dcfb5edcc5d47b87383/engine/access/rest/routes/subscribe_events_test.go#L73